contributor-docs/code-change-guide.md

```diff
@@ -0,0 +1,518 @@
+```
+...
+BUILD SUCCESSFUL in 2m 32s
+96 actionable tasks: 9 executed, 87 up-to-date
+3:41:06 PM: Execution finished 'wordCount'.
+```
+
+In addition, the following text appears in the output file:
+
+```shell
+
+$ head /tmp/output.txt*
+==> /tmp/output.txt-00000-of-00003 <==
+should: 38
+bites: 1
+depraved: 1
+gauntlet: 1
+battle: 6
+sith: 2
+cools: 1
+natures: 1
+hedge: 1
+words: 9
+
+==> /tmp/output.txt-00001-of-00003 <==
+elements: 1
+Advise: 2
+fearful: 2
+towards: 4
+ready: 8
+pared: 1
+left: 8
+safe: 4
+canst: 7
+warrant: 2
+
+==> /tmp/output.txt-00002-of-00003 <==
+chanced: 1
+...
+```
+
+*What does this command do?*
+
+This command compiles the beam SDK and the WordCount pipeline, a Hello-world program for
+data processing, then runs the pipeline on the Direct Runner.
+
+## Run a unit test
+
+This section explains how to run unit tests locally after you make a code change in the Java SDK (for example, in `sdks/java/io/jdbc`).
+
+Tests are under the `src/test/java` folder of each project. Unit tests have the filename `.../**Test.java`. Integration tests have the filename `.../**IT.java` for integration tests.
+
+* To run all unit tests under a project, use the following command:
+  ```
+  ./gradlew :sdks:java:harness:test
+  ```
+  Find the  JUnit report in an HTML file in the file path `<invoked_project>/build/reports/tests/test/index.html`.
+
+* To run a specific test, use the following commands:
+  ```
+  ./gradlew :sdks:java:harness:test --tests org.apache.beam.fn.harness.CachesTest
+  ./gradlew :sdks:java:harness:test --tests *CachesTest
+  ./gradlew :sdks:java:harness:test --tests *CachesTest.testClearableCache
+  ```
+
+* To run tests using IntelliJ, click the ticks to run a whole test class or a specific test. You can set breakpoints to debug the test.
+  <img width="452" alt="image" src="https://github.com/apache/beam/assets/8010435/7ae2a65c-a104-48a2-8bad-ff8c52dd1943">
+
+* These steps don't apply to `sdks:java:core` tests. Instead, invoke the unit tests by using `:runners:direct-java:needsRunnerTest`. Java core doesn't depend on a runner. Therefore, unit tests that run a pipeline require the Direct Runner.
+
+  Reason: Java core does not depend on a runner. For unit test that executes a pipeline, it needs direct runner.
+
+To run integration tests, use the Direct Runner.
+
+## Run integration tests (*IT.java)
+
+Integration tests use [`TestPipeline`](https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/testing/TestPipeline.java). Set options by using `TestPipelineOptions`.
+
+Integration tests differ from standard pipelines in the following ways:
+* By default, they block on run (on TestDataflowRunner).
+* They have default timeout of 15 minutes.
+* The pipeline options are set in the system property `beamTestPipelineOptions`.
+
+Note the final different, because you need to configure the test by setting `-DbeamTestPipelineOptions=[...]`. This property is where you set the runner to use.
```

Comment:
```suggestion
Note the final difference, because you need to configure the test by setting `-DbeamTestPipelineOptions=[...]`. This property is where you set the runner to use.
```